tools/Google.Cloud.Tools.VersionCompat.Tests/ClassTest.cs

```diff
@@ -0,0 +1,136 @@
+    namespace InterfaceAdded.B { public abstract class C : I1 { } }
+
+    namespace InterfaceRemoved.A { public abstract class C : I1 { } }
+    namespace InterfaceRemoved.B { public abstract class C { } }
+
+    namespace NonExportedInterfaceAdded.A { public abstract class C { } }
+    namespace NonExportedInterfaceAdded.B { public abstract class C : I2 { } }
+
+    namespace NonExportedInterfaceRemoved.A { public abstract class C : I2 { } }
+    namespace NonExportedInterfaceRemoved.B { public abstract class C { } }
+
+    namespace GenericConstraintAdded1.A { public class C<A> { } }
+    namespace GenericConstraintAdded1.B { public class C<A> where A : new() { } }
+
+    namespace GenericConstraintAdded2.A { public class C<A> { } }
+    namespace GenericConstraintAdded2.B { public class C<A> where A : class { } }
+
+    namespace GenericConstraintAdded3.A { public class C<A> { } }
+    namespace GenericConstraintAdded3.B { public class C<A> where A : struct { } }
+
+    namespace GenericConstraintAdded4.A { public class C<A> { } }
+    namespace GenericConstraintAdded4.B { public class C<A> where A : IList<A> { } }
+
+    namespace GenericConstraintRemoved1.A { public class C<A> where A : new() { } }
+    namespace GenericConstraintRemoved1.B { public class C<A> { } }
+
+    namespace GenericConstraintRemoved2.A { public class C<A> where A : class { } }
+    namespace GenericConstraintRemoved2.B { public class C<A> { } }
+
+    namespace GenericConstraintRemoved3.A { public class C<A> where A : struct { } }
+    namespace GenericConstraintRemoved3.B { public class C<A> { } }
+
+    namespace GenericConstraintRemoved4.A { public class C<A> where A : IList<A> { } }
+    namespace GenericConstraintRemoved4.B { public class C<A> { } }
+}
+
+namespace Google.Cloud.Tools.VersionCompat.Tests
+{
+    public class ClassTest : TestBase
+    {
+        [Fact] public void NoChange() => TestNone();
+        // TOOD: Enable this test after ctors are dealt with properly.
+        // [Fact] public void ToStatic() => TestMajor(Cause.ClassModifierChanged);
+        [Fact] public void ToStaticNoCtor() => TestNone();
+        [Fact] public void ToAbstract() => TestMajor(Cause.ClassModifierChanged);
+        [Fact] public void ToAbstractNoCtor() => TestNone();
+        [Fact] public void ToSealed() => TestMajor(Cause.ClassModifierChanged);
+        // TOOD: Enable this test after ctors are dealt with properly.
+        // [Fact] public void FromStatic() => TestMinor(Cause.ClassModifierChanged);
+        [Fact] public void FromStaticNoCtor() => TestNone();
+        [Fact] public void FromAbstract() => TestMinor(Cause.ClassModifierChanged);
+        [Fact] public void FromAbstractNoCtor() => TestNone();
+        [Fact] public void FromSealed() => TestMinor(Cause.ClassModifierChanged);
+        [Fact] public void FromSealedNoCtor() => TestNone();
+        [Fact] public void BaseTypeFromObjectOk() => TestMinor(Cause.ClassBaseClassChanged);
```

Comment:
I think we'll need documentation justifying a fair number of these. That can come later, of course.

Reply:
Yes, completely agree. We need to be able to review them all easily and check everything is correct. Sometimes things really aren't very obvious!!